projects/igniteui-angular/src/lib/date-range/igx-date-range.component.ts

```diff
@@ -0,0 +1,696 @@
+     *  this.dateRange.selectRange(today, inFiveDays);
+     * }
+     * ```
+     */
+    public selectRange(startDate: Date, endDate?: Date): void {
+        endDate = endDate ?? startDate;
+        const dateRange = [startDate, endDate];
+        this.calendar.selectDate(dateRange);
+        this.handleSelection(dateRange);
+    }
+
+    /** @hidden @internal */
+    public writeValue(value: DateRange): void {
+        this.value = value;
+    }
+
+    /** @hidden @internal */
+    public registerOnChange(fn: any): void {
+        this._onChangeCallback = fn;
+    }
+
+    /** @hidden @internal */
+    public registerOnTouched(fn: any): void { }
+
+    /** @hidden @internal */
+    get separatorClass(): string {
+        return this.getComponentDensityClass('igx-date-range__label');
+    }
+
+    /** @hidden */
+    public ngAfterViewInit(): void {
+        switch (this.mode) {
+            case InteractionMode.DropDown:
+                this.attachOnKeydown();
+                this.applyFocusOnClose();
+                this.subscribeToDateEditorEvents();
+                break;
+            case InteractionMode.Dialog:
+                this.applyFocusOnClose();
+                break;
+        }
+        this.configPositionStrategy();
+        this.configOverlaySettings();
+
+        const subsToClicked = () => {
+            this.$toggleClickNotifier.next();
+            this.toggleComponents.forEach(toggle => {
+                toggle.clicked.pipe(takeUntil(this.$toggleClickNotifier)).subscribe(() => this.open());
+            });
+        };
+
+        this.toggleComponents.changes.pipe(takeUntil(this.$destroy)).subscribe(() => subsToClicked());
+        subsToClicked();
+    }
+
+    /** @hidden @internal */
+    public ngOnChanges(changes: SimpleChanges): void {
+        if (changes['locale']) {
+            this.inputFormat = DatePickerUtil.getDefaultInputFormat(this.locale || 'en') || DEFAULT_INPUT_FORMAT;
+        }
+    }
+
+    /** @hidden @internal */
+    public ngOnDestroy(): void {
+        this.$destroy.next();
+        this.$destroy.complete();
+        this.$toggleClickNotifier.next();
+        this.$toggleClickNotifier.complete();
+    }
+
+    /** @hidden @internal */
+    public handleOpening(event: CancelableEventArgs): void {
+        this.onOpening.emit(event);
+    }
+
+    /** @hidden @internal */
+    public handleOpened(): void {
+        requestAnimationFrame(() => {
+            this.calendar.daysView.focusActiveDate();
+            this.onOpened.emit({ owner: this });
+        });
+    }
+
+    /** @hidden @internal */
+    public handleClosing(event: CancelableEventArgs): void {
+        this.onClosing.emit(event);
+    }
+
+    /** @hidden @internal */
+    public handleClosed(): void {
+        if (this.value && !this.value.end) {
+            this.value = { start: this.value.start, end: this.value.start };
+            this.onClosed.emit({ owner: this });
```

Comment:
this should not be in the `if` statement
now if we have an end value event will not be triggered